### DISCUSSION TITLE
Better output for `+list-colors`

### DISCUSSION THREAD

**Discussion by anund** [ghostty-org/ghostty#3064] [Feature Requests, Ideas] [answered]
Following on from the work in https://github.com/ghostty-org/ghostty/pull/2052 it would be nice to add color and additional formatting to `+list-colors`. At the very least printing the color needs output a colored result.

Here's an attempt:
![2024-12-22-181102_1159x249_scrot](https://github.com/user-attachments/assets/1ffe3986-1a5f-488a-a526-47e654d10be3)

- the name sorted output has groupings of colors that print more nicely down a column than across a row
- there are a lot of grey colors, they can't really be grouped to not break across from one column to the other
- terminal width controls the number of columns, which controls when a color group (color1, color2) might be split across columns 
- the ascii `■` prints across 2 columns with the font in the screenshot, overlapping the following space. Is there something better? 
- no idea how this looks on windows
- the longest color name is `light goldenrod yellow` leading to the space between names and `=`
- a `--plain` flag and tty checking are in place following the way `+list-keybinds` works

Suggestions for formatting?

**Comment by anund** [comment]
Code is here: https://github.com/ghostty-org/ghostty/compare/main...anund:ghostty-reforked:pretty_print_colors

**Reply by furtidev** [reply]
Hey, your branch seems inaccessible.

**Reply by anund** [reply]
fixed

**Reply by anund** [reply]
current branch uses `██` rather than the `■` in the screenshot

**Comment by mitchellh** [comment, marked as answer]
#8386 accepted
